plugins/outputs/warp10/warp10.go

```diff
@@ -0,0 +1,204 @@
+package warp10
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+const (
+	defaultClientTimeout = 15 * time.Second
+)
+
+// Warp10 output plugin
+type Warp10 struct {
+	Prefix  string
+	WarpURL string
+	Token   string
+	Timeout internal.Duration `toml:"timeout"`
+	client  *http.Client
+}
+
+var sampleConfig = `
+  # prefix for metrics class Name
+  prefix = "Prefix"
+  ## POST HTTP(or HTTPS) ##
+  # Url name of the Warp 10 server
+  warp_url = "WarpUrl"
+  # Token to access your app on warp 10
+  token = "Token"
+  # Warp 10 query timeout, by default 15s
+  timeout = "15s"
+`
+
+// MetricLine Warp 10 metrics
+type MetricLine struct {
+	Metric    string
+	Timestamp int64
+	Value     string
+	Tags      string
+}
+
+func (w *Warp10) createClient() (*http.Client, error) {
+	client := &http.Client{
+		Timeout: w.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+// Connect to warp10
+func (w *Warp10) Connect() error {
+	client, err := w.createClient()
+	if err != nil {
+		return err
+	}
+
+	if w.Timeout.Duration == 0 {
+		w.Timeout.Duration = defaultClientTimeout
+	}
+
+	w.client = client
+	return nil
+}
+
+// GenWarp10Payload compute Warp 10 metrics payload
+func (w *Warp10) GenWarp10Payload(metrics []telegraf.Metric, now time.Time) string {
+	collectString := make([]string, 0)
+	index := 0
+	for _, mm := range metrics {
+
+		for _, field := range mm.FieldList() {
+
+			metric := &MetricLine{
+				Metric:    fmt.Sprintf("%s%s", w.Prefix, mm.Name()+"."+field.Key),
+				Timestamp: now.UnixNano() / 1000,
+			}
+
+			metricValue, err := buildValue(field.Value)
+			if err != nil {
+				log.Printf("Warp: %s\n", err.Error())
+				continue
+			}
+			metric.Value = metricValue
+
+			tagsSlice := buildTags(mm.Tags())
```

Comment:
Sorry I missed this before, we will want to use TagList() here for the same reasons we switched to FieldList(), its quite a bit more performant (and we will be removing this method soon).